.github/workflows/pytest.yml

```diff
@@ -0,0 +1,28 @@
+name: pytest
+
+on:
+    pull_request:
+    push:
+        branches: [main]
+
+jobs:
+    pytest:
+        runs-on: ubuntu-latest
+        strategy:
+            matrix:
+                python-version: [3.7, 3.8, 3.9]
+
+        steps:
+            - uses: actions/checkout@v2
+            - name: Set up Python ${{ matrix.python-version }}
+              uses: actions/setup-python@v2
+              with:
+                  python-version: ${{ matrix.python-version }}
+            - name: Install dependencies
+              run: |
+                  python -m pip install --upgrade pip
+                  pip install pytest
+                  if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
```

Comment:
I'd also add `pip install dev-requirements.txt` because that's where `pytest` lives and we can remove the line above to explicitly install it in the action.

Reply:
alright I'll add that